presto-main/src/main/java/com/facebook/presto/cost/AggregationStatsRule.java

```diff
@@ -76,27 +94,59 @@ public static PlanNodeStatsEstimate groupBy(PlanNodeStatsEstimate sourceStats, C
             }));
         }
 
-        double rowsCount = 1;
+        double rowsCount = getDefaultAggregateSelectivityCoefficient(session);
+        double totalSize = 0;
         for (VariableReferenceExpression groupByVariable : groupByVariables) {
             VariableStatsEstimate symbolStatistics = sourceStats.getVariableStatistics(groupByVariable);
             int nullRow = (symbolStatistics.getNullsFraction() == 0.0) ? 0 : 1;
             rowsCount *= symbolStatistics.getDistinctValuesCount() + nullRow;
+            if (!isNaN(symbolStatistics.getAverageRowSize())) {
+                totalSize += symbolStatistics.getAverageRowSize() * rowsCount;
```

Comment:
this seems incorrect since the rowsCount will change as we go through each column.  probably we want to add all the average row sizes per-column and then at the end multiply that by whatever row count we've ended up that.

Reply:
fixed